File: analysis/data/model_inputs/JPE_replication/extract_from_mle_converged.py
import scipy.io

# Load the .mat file
mat_data = scipy.io.loadmat('analysis/data/model_inputs/JPE_replication/mle_converged.mat')

# Now mat_data is a dictionary containing the variables from the .mat file
print(mat_data.keys())
sol_mle = mat_data['sol_mle']
mp_mle = mat_data['mp_mle']
Avar_mle = mat_data['Avar_mle']

# prices
prices=sol_mle[0][0][0]

# scrap probs: 
sol_mle[0][0][2][0][0][2]



breakpoint()

File: analysis/data/model_inputs/JPE_replication/extract_from_mle_converged.m

%% 
sol_mle=load('mle_converged.mat');


%% save output for later: 

% scrap probs
csvwrite('scrap_probabilities_model.csv',cell2mat(sol_mle.ccp_scrap_tau'))

% prices:
csvwrite('used_car_prices_model.csv', sol_mle.p)

% EV terms
csvwrite('ev_terms_model.csv', cell2mat(sol_mle.ev_tau'))

% estimates
save('mp_mle_model.mat', '-struct', 'mp_mle');

%
to_export = ['acc_0', 'acc_a', 'sigma_s', 'mum', 'psych_transcost', 'psych_transcost_nocar', 'tc_sale', 'tc_sale_even', 'u_0', 'u_a']
for i = 1:length(to_export)
    save('mp_mle_model.mat', '-struct', mp_mle.(to_export{i}));
end